text/0148-sampling-probability.md

```diff
@@ -0,0 +1,744 @@
+procedure must be _statistically unbiased_, a term meaning that the
+process is required to give equal consideration to all possible
+outcomes.
+
+#### Sampling and variance
+
+The use of unbiased sampling outlined above makes it possible to
+estimate the population total for arbitrary subsets of the sample, as
+every individual sample has been independently assigned an adjusted
+count.
+
+There is a natural relationship between statistical bias and variance.
+Approximate counting comes with variance, a matter of fact which can
+be controlled for by the sample size.  Variance is unavoidable in an
+unbiased sample, but it vanishes when you have enough data.
+
+Although this makes it sound like small sample sizes are a problem,
+due to expected high variance, this is just a limitation of the
+technique.  When variance is high, use a larger sample size.
+
+An easy approach for lowering variance is to aggregate sample frames
+together across time.  For example, although the estimates drawn from
+a one-minute sample may have high variance, combining an hour of
+one-minute sample frames into an aggregate data set is guaranteed to
+lower variance.  It must, because the data remains unbiased.
+
+### Conveying the sampling probability
+
+Some possibilities for encoding the adjusted count or inclusion
+probability are discussed below, depending on the circumstances and
+the protocol.
+
+There are several ways of encoding this information:
+
+- as a dedicated field in an OTLP protobuf message
+- as a non-descriptive Attribute in an OTLP Span, Metric, or Log
+- without any dedicated field.
+
+#### Encoding adjusted count
+
+We can encode the adjusted count directly as a floating point or
+integer number in the range [0, +Inf).  This is a conceptually easy
+way to understand sampling because larger numbers mean greater
+representivity.
+
+#### Encoding inclusion probability
+
+We can encode the inclusion probability directly as a floating point
+number in the range [0, 1).  This is typical of the Statsd format,
+where each line includes an optional probability.  In this context,
+the probability is also commonly referred to as a "sampling rate".  In
+this case, smaller numbers mean greater representivity.
+
+#### Encoding negative base-2 logarithm of inclusion probability
+
+We can encode the negative base-2 logarithm of inclusion probability.
+This restricts inclusion probabilities to powers of two and allows the
+use of small non-negative integers to encode power-of-two adjusted
+counts.  In this case, larger numbers mean exponentially greater
+representivity.
+
+#### Multiply the adjusted count into the data
+
+When the data itself carries counts, such as for the Metrics Sum and
+Histogram points.
+
+This technique is less desirable because while it preserves the
+expected value of the count or sum, the data loses information about
+variance.  This may also lead to rounding errors, when adjusted counts
```

Comment:
+1, all the folks that I know + who made such choice, regret for it. This is especially painful when the metrics are used for alerting, the information loss about variance makes it hard to tell the confidence level of the alert.

Reply:
:+1: